crates/motoko/src/lib/vm.rs

```diff
@@ -2469,7 +2470,12 @@ fn nonempty_stack_cont<A: Active>(active: &mut A, v: Value_) -> Result<Step, Int
                 // do projection, representing function with special value.
                 Ok(Step {})
             }
-            _ => type_mismatch!(file!(), line!()),
+            v => Err(type_mismatch_!(
+                file!(),
+                line!(),
+                "dot-operator-is-matching-operand",
+                format!("{:?} @ {}", v, active.cont_source())
+            )),
```

Comment:
@rvanasa It took some guesswork to figure out, based on the position in the file and knowing what to expect there, which file in `base` is actually being referenced when I got this message popping out.

So in conclusion, I guess we need to finally use "filenames" in our `Source` information, at least for this use case (and probably in general going forward, since we have imports, the base library, etc.)